lerobot/scripts/online_training_helpers.py

```diff
@@ -0,0 +1,292 @@
+        self._buffer_capacity = buffer_capacity
+        data_spec = self._make_data_spec(data_spec, buffer_capacity)
+        os.makedirs(write_dir, exist_ok=True)
+        self._data = {
+            k: _make_memmap_safe(
+                filename=Path(write_dir) / k,
+                dtype=v["dtype"] if v is not None else None,
+                mode="r+" if (Path(write_dir) / k).exists() else "w+",
+                shape=tuple(v["shape"]) if v is not None else None,
+            )
+            for k, v in data_spec.items()
+        }
+
+    @property
+    def delta_timestamps(self) -> dict[str, np.ndarray] | None:
+        return self._delta_timestamps
+
+    def set_delta_timestamps(self, value: dict[str, list[float]] | None):
+        """Set delta_timestamps converting the values to numpy arrays.
+
+        The conversion is for an optimization in the __getitem__. The loop is much slower if the arrays
+        need to be converted into numpy arrays.
+        """
+        if value is not None:
+            self._delta_timestamps = {k: np.array(v) for k, v in value.items()}
+        else:
+            self._delta_timestamps = None
+
+    def _make_data_spec(self, data_spec: dict[str, Any], buffer_capacity: int) -> dict[str, dict[str, Any]]:
+        """Makes the data spec for np.memmap."""
+        if any(k.startswith("_") for k in data_spec):
+            raise ValueError(
+                "data_spec keys should not start with '_'. This prefix is reserved for internal logic."
+            )
+        preset_keys = {"index", "frame_index", "episode_index", "timestamp"}
+        if len(intersection := set(data_spec).intersection(preset_keys)) > 0:
+            raise ValueError(
+                f"data_spec should not contain any of {preset_keys} as these are handled internally. "
+                f"The provided data_spec has {intersection}."
+            )
+        data_spec = {
+            # _next_index will be a pointer to the next index that we should start filling from when we add
+            # more data.
+            "_next_index": {"dtype": np.dtype("int64"), "shape": (1,)},
+            # Since the memmap is initialized with all-zeros, this keeps track of which indices are occupied
+            # with real data rather than the dummy initialization.
+            "_occupancy_mask": {"dtype": np.dtype("?"), "shape": (buffer_capacity,)},
+            "index": {"dtype": np.dtype("int64"), "shape": (buffer_capacity,)},
+            "frame_index": {"dtype": np.dtype("int64"), "shape": (buffer_capacity,)},
+            "episode_index": {"dtype": np.dtype("int64"), "shape": (buffer_capacity,)},
+            "timestamp": {"dtype": np.dtype("float64"), "shape": (buffer_capacity,)},
+            **{
+                k: {"dtype": v["dtype"], "shape": (buffer_capacity, *v["shape"])}
+                for k, v in data_spec.items()
+            },
+        }
+        return data_spec
+
+    def add_data(self, data: dict[str, np.ndarray]):
```

Comment:
In our use case, is it frame, a trajectory, an episode?

From my understanding, it's an episode, but we can support all three. Could be nice to add a comment maybe?